src/SerilogWeb.Classic/Classic/ApplicationLifecycleModule.cs

```diff
@@ -184,9 +212,25 @@ public void Init(HttpApplication application)
             };
         }
 
-        static bool ShouldLogFormData => Logger.IsEnabled(_formDataLoggingLevel) &&
-            (LogPostedFormData == LogPostedFormDataOption.Always ||
-                (LogPostedFormData == LogPostedFormDataOption.OnlyOnError && HttpContext.Current.Response.StatusCode >= 500));
+        static Func<HttpContext, bool> FormLoggingStrategy
+        {
+            get
+            {
+                switch (_logPostedFormData)
+                {
+                    case LogPostedFormDataOption.Never:
+                        return context => false;
```

Comment:
Alternatively (instead of making this a method), could the constant cases return a cached delegate?

```
case LogFormDataOption.Always:
   return AlwaysTrue;
```

where `AlwaysTrue` is a `static readonly` field `context => true`. Creates a (minimal) bit of garbage on each request otherwise.